app.js

```diff
@@ -26,16 +27,33 @@ winston.add(winston.transports.File, { filename: logFile });
 const exitnodes = ['45.34.140.42'];
 app.exitnodes = exitnodes;
 
-app.get('/', function(req, res) {
-  mjs.get('alivejson', function(err, v) {
-    //TODO Handle the error!
-    let msg = util.messageFromCacheData(v);
-    //TODO Might make sense to do
-    // if (msg.error) { res.status(404); }
-    // ...but the issue isn't that the client specified a bad URI, so...
-    res.render('index', {value: msg});
+/**
+ * Get a data object from memcache by key
+ * @param {String} key - memcache key to read
+ * @returns {Object} - parsed Object from memcache 
+ */
+async function getCacheData(key) {
+  const { value } = await mjs.get(key)
+  const data = JSON.parse(value)
+  return data
+}
+
+/**
+ * wrap a function that returns promise and return an express middleware
+ * @param {Function} fn - A function that accepts (req, res, next) and returns promise 
+ * @returns {Function} an express middleware (request handler)
+ */
+const asyncMiddleware = fn => (req, res, next) => {
+  Promise.resolve(fn(req, res, next)).catch(next);
```

Comment:
Could this be: `fn(req, res, next).catch(next)` instead of `Promise.resolve(fn(req, res, next)).catch(next)` ? Is `Promise.resolve` there in case `fn` doesn't actually return a promise?